### PR TITLE
Fix/dynamic id validation

### PR DESCRIPTION
**Pull request by y-golde**
added dynamic validation for characters and length of identification ( depending on type )

### CODE REVIEW

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningPersonal.tsx [hunk 1]
```diff
@@ -109,10 +109,13 @@ const ContactQuestioningPersonal: React.FC<Props> = (
                             defaultValue={
                                 interactedContact.identificationNumber
                             }
+                            max={9}
                             render={(props) => {
+                                console.log(props);
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningPersonal.tsx:114]: delete

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningPersonal.tsx [hunk 2]
```diff
@@ -109,10 +109,13 @@ const ContactQuestioningPersonal: React.FC<Props> = (
...
                                 return (
                                     <IdentificationTextField
                                         {...props}
+                                        isPassport={isPassport}
                                         disabled={shouldIdDisable}
                                         testId="identificationNumber"
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningPersonal.tsx:120]: single quotes

File: client/src/commons/IdentificationTextField/IdentificationTextField.tsx [hunk 3]
```diff
@@ -1,23 +1,18 @@
...
-const errorMessage = 'הוכנס תו לא וולידי';
-const maxLengthErrorMessage = `השדה יכול להכיל ${maxIdentificationLength} תווים בלבד`;
-
-export const stringAlphanum = yup
-  .string()
-  .matches(generalIdentificationValidation, errorMessage)
-  .max(maxIdentificationLength, maxLengthErrorMessage);
-
 const IdentificationTextField = (props : Props) => {
+  const { isPassport } = props;
```
Comment (ItayBenMoshe) [inline client/src/commons/IdentificationTextField/IdentificationTextField.tsx:7]: enter

File: client/src/Utils/Schemas/identification.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import * as yup from 'yup';
+
+import { passportValidation, passportMaxIdentificationLength , idLength , idBasicValidation} from 'Utils/auxiliaryFunctions/auxiliaryFunctions';
+
+const errorMessage = 'הוכנס תו לא וולידי';
```
Comment (barnahum8) [inline client/src/Utils/Schemas/identification.ts:5]: I would change 'וולידי' to 'חוקי'

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningPersonal.tsx (hunk 1 above)
Comment (barnahum8) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningPersonal.tsx:114]: remove console log

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningPersonal.tsx (hunk 2 above)
Comment (y-golde) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningPersonal.tsx]: fixed [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1116/commits/58099a078aee6c73f9297019cfc108b9fae024e7)

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningPersonal.tsx (hunk 1 above)
Comment (y-golde) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningPersonal.tsx]: fixed [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1116/commits/58099a078aee6c73f9297019cfc108b9fae024e7)

File: client/src/commons/IdentificationTextField/IdentificationTextField.tsx (hunk 3 above)
Comment (y-golde) [reply client/src/commons/IdentificationTextField/IdentificationTextField.tsx]: fixed [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1116/commits/58099a078aee6c73f9297019cfc108b9fae024e7)

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningPersonal.tsx (hunk 1 above)
Comment (y-golde) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningPersonal.tsx]: fixed [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1116/commits/58099a078aee6c73f9297019cfc108b9fae024e7)

File: client/src/Utils/Schemas/identification.ts (hunk 4 above)
Comment (y-golde) [reply client/src/Utils/Schemas/identification.ts]: fixed [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1116/commits/7047524a85644e5143476711afe92d201be6bfdc)

**Review by ItayBenMoshe — APPROVED** [review]

**Review by barnahum8 — APPROVED** [review]

**Pull request merged**